great_expectations/checkpoint/checkpoint.py

```diff
@@ -400,3 +408,179 @@ def _get_batches_to_validate(self, batches):
+        "name": "store_evaluation_params",
+        "action": {"class_name": "StoreEvaluationParametersAction"},
+    }
+    UPDATE_DATA_DOCS = {
+        "name": "update_data_docs",
+        "action": {"class_name": "UpdateDataDocsAction", "site_names": None},
+    }
+
+    @staticmethod
+    def build_slack_action(webhook, notify_on, notify_with):
+        return {
+            "name": "slack_notification",
+            "action": {
+                "class_name": "SlackNotificationAction",
+                "slack_webhook": webhook,
+                "notify_on": notify_on,
+                "notify_with": notify_with,
+            },
+        }
+
+
+class SimpleCheckpoint(Checkpoint):
+    """
+    SimpleCheckpoint is an easy to use Checkpoint with enjoyable defaults.
+
+    This Checkopint performs the following actions by default:
+    1. store the validation result
+    2. store evaluation parameters
+    3. update all data docs sites
+
+    When configured, this Checkpoint can also send a slack message.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        data_context,
+        site_names: Optional[Union[str, List[str]]] = "all",
+        slack_webhook: Optional[str] = None,
+        notify_on: Optional[str] = "all",
+        notify_with: Optional[Union[str, List[str]]] = "all",
+    ):
+        # TODO nice docstring explaining arguments and example usages
+        self._site_names = site_names
+        self._validate_site_names()
+
+        self._notify_on = notify_on
+        self._validate_notify_on()
+
+        self._notify_with = notify_with
+        self._validate_notify_with()
+
+        self._slack_webhook = slack_webhook
+        self._validate_slack_webhook()
+        self._validate_slack_configuration()
+
+        checkpoint_config = self._build_checkpoint_config()
+        super().__init__(name, data_context, checkpoint_config)
+
+    @property
+    def notify_on(self) -> Optional[str]:
+        return self._notify_on
+
+    @property
+    def notify_with(self) -> Union[str, List[str], None]:
+        return self._notify_with
+
+    @property
+    def site_names(self) -> Union[str, List[str], None]:
+        return self._site_names
+
+    @property
+    def slack_webhook(self) -> Optional[str]:
+        return self._slack_webhook
+
+    def _build_checkpoint_config(self) -> Dict:
+        action_list = self._default_action_list()
+        if self.site_names:
+            action_list = self._add_update_data_docs_action(action_list)
+        if self.slack_webhook:
+            action_list = self._add_slack_action(action_list)
+        checkpoint_config = {
+            "config_version": 1.0,
+            "class_name": "SimpleCheckpoint",
+            "action_list": action_list,
+        }
+        logger.debug(
+            f"SimpleCheckpoint built this CheckpointConfig: {json.dumps(checkpoint_config, indent=4)}"
+        )
+        return checkpoint_config
+
+    def _default_action_list(self) -> List[Dict]:
```

Comment:
Just curious - why a private method and not just a constant?

Reply:
Good question! I did that first, and since I'm mutating them it ended up being messy with a bunch of `.copy()`s strewn about so a method returns a new object for each invocation.